From: Jiang Xin <worldhello.net@gmail.com>
Subject: [PATCH] When delete branch, delete it's remote couterparts

When delete topic branch, delete it's remote conterparts using:

    git push origin :refs/heads/t/topic
    git push origin :refs/top-bases/t/topic

Signed-off-by: Jiang Xin <worldhello.net@gmail.com>

---
 tg-delete.sh |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tg-delete.sh b/tg-delete.sh
index e1eea17..41df1ca 100644
--- a/tg-delete.sh
+++ b/tg-delete.sh
@@ -46,4 +46,16 @@ baserev="$(git rev-parse --verify "refs/top-bases/$name" 2>/dev/null)" ||
 git update-ref -d "refs/top-bases/$name" "$baserev"
 [ -z "$branchrev" ] || git update-ref -d "refs/heads/$name" "$branchrev"
 
+## Wipe out remote branches
+
+if git rev-parse --verify -q "refs/remotes/$base_remote/top-bases/$name" >/dev/null; then
+	info "Delete remote branch: refs/top-bases/$name"
+	git push $base_remote :"refs/top-bases/$name"
+fi
+
+if git rev-parse --verify -q "refs/remotes/$base_remote/$name" >/dev/null; then
+	info "Delete remote branch: refs/heads/$name"
+	git push $base_remote :"refs/heads/$name"
+fi
+
 # vim:noet
-- 
tg: (d279e29..) t/delete-remote-branch (depends on: tgmaster)
